### PR TITLE
More Typings

### PR DESCRIPTION
**Pull request by ChiefArug**
### Description 
Moar typings 🎉 

#### Other details 
In doing this I have noticed a lot of seemingly redundant code, particularly around BlockWrapper and BlockPredicate, but have not touched that for fear of breaking addons. I haven't bothered documenting it though.

### CODE REVIEW

File: common/src/main/java/dev/latvian/mods/kubejs/bindings/ItemWrapper.java [hunk 1]
```diff
@@ -13,62 +14,79 @@
...
 	static boolean exists(ResourceLocation id) {
 		return KubeJSRegistries.items().contains(id);
 	}
 
+	@Info("""
+		Checks if the passed in object is an ItemStack.
+		Note that this does not mean it will not function as an ItemStack if passed to something that requests one.
+		""")
 	static boolean isItem(@Nullable Object o) {
 		return o instanceof ItemStackJS;
```
Comment (MaxNeedsSnacks) [inline common/src/main/java/dev/latvian/mods/kubejs/bindings/ItemWrapper.java:91]: just noticed this error while going through to review this, this should be instanceof ItemStack, shouldn't it?

Comment (ChiefArug) [reply common/src/main/java/dev/latvian/mods/kubejs/bindings/ItemWrapper.java:91]: ItemStackMixin does implement ItemStackJS, but I'm not sure if that gets added to ItemStack as part of the mixin.

**Comment by Prunoideae** [comment]
It would be better if documents were in markdown format.

File: common/src/main/java/dev/latvian/mods/kubejs/bindings/ItemWrapper.java (hunk 1 above)
Comment (MaxNeedsSnacks) [reply common/src/main/java/dev/latvian/mods/kubejs/bindings/ItemWrapper.java:91]: it implemented ItemStack__KJS__ not ItemStackJS; ItemStackJS is just the leftovers of the item stack wrapper iirc

Comment (ChiefArug) [reply common/src/main/java/dev/latvian/mods/kubejs/bindings/ItemWrapper.java:91]: Oh whoops, yeah. Do you want me to fix that in this PR?

**Comment by ChiefArug** [comment]
Do the docs need formatting at all?

**Comment by Prunoideae** [comment]
VSCode renders docs as markdown by default, though.

**Comment by ChiefArug** [comment]
@LatvianModder are you all good with having @Info use markdown? Or should it be left as plain text.
Also, should I include examples for the `of` methods (ie ItemWrapper#of, TextWrapper#of(

**Comment by ChiefArug** [comment]
I'm marking this as ready for merging now, but may continue to push commits if I feel like documenting anything else.

**Pull request merged**